proxy/hdrs/HTTP.cc

```diff
@@ -1088,6 +1088,11 @@ http_parser_parse_req(HTTPParser *parser, HdrHeap *heap, HTTPHdrImpl *hh, const
     }
     url_end += 1;
 
+    // Make sure we didn't pickup odd characters in the url between the method and version
+    if (!url_is_compliant(url_start, url_end)) {
```

Comment:
The same check will be done later when `url_parse` is called, if `strict_uri_parsing` is `true`. In other words, there might be a case we want to accept an invalid URL? Adding this check here effectively makes the flag void.